contracts/protocols/mainnet/erc4626/main.sol

```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+import "./interfaces.sol";
+
+contract Resolver {
+    struct VaultData {
+        bool isToken;
+        string name;
+        string symbol;
+        uint256 decimals;
+        address asset;
+        uint256 totalAssets;
+    }
+
+    function getVaultDetails(address vaultAddress) public view returns (VaultData memory) {
+        VaultData memory _vaultData;
+        VaultInterface vaultToken = VaultInterface(vaultAddress);
+        bool isToken = true;
+
+        try vaultToken.symbol() {} catch {
+            isToken = false;
+        }
+
+        try vaultToken.name() {} catch {
+            isToken = false;
+        }
+
+        try vaultToken.decimals() {} catch {
+            isToken = false;
+        }
+
+        _vaultData = VaultData(
+            isToken,
+            vaultToken.name(),
+            vaultToken.symbol(),
+            vaultToken.decimals(),
+            vaultToken.asset(),
+            vaultToken.totalAssets()
+        );
+
+        return _vaultData;
+    }
+
+    function getPosition(address owner, address vaultAddress) public view returns (uint256, uint256) {
```

Comment:
Similar to how we do on ERC20 resolver. Also add a function as `getPositions(address owner, address[] vaultAddresses)` to fetch mutiple vault's data at once.